include/selene/BaseFun.h

```diff
@@ -2,7 +2,7 @@
 
 #include "exotics.h"
 #include <exception>
-#include "exception.h"
+#include "ExceptionHandler.h"
```

Comment:
Renamed `exception.h` to `ExceptionHandler.h` to emphasize the difference between the handling of internally thrown exceptions and the newly user visible exception types in `ExceptionTypes.h`